LIPS/lip-23.md

```diff
@@ -0,0 +1,143 @@
+---
+lip: 23
+title: Negative rebase sanity check with second opinion
+status: Proposed
+author: Alexey Potapkin, Greg Shestakov, Eugene Pshenichnyi
+discussions-to: https://research.lido.fi/t/
+created: 2024-04-17
+updated: 2024-05-22
+---
+
+# LIP-23: Negative rebase sanity check with second opinion
+
+## Abstract
+
+Improve the safety check for the accounting report in the case of a negative rebase, reducing the possible impact size, but with the requirement for a second opinion for extreme cases. As a result impact size of current sanity check reduced by 26 times to 0.19% per day.
+
+## Motivation
+
+The `AccountingOracle` contract is a fundamental component of the Lido protocol that delivers, amongst other data, the aggregate of all Lido validators' Beacon Chain balances (`clBalance` in the report) to the protocol, thereby facilitating the daily rebase of the stETH token. The accuracy of this value is critical to the integrity of the protocol, and it relies on a committee of independently operated Oracle daemons in a 5-of-9 configuration. The protocol could be harmed if this committee is compromised, malfunctions, or colludes. This risk is acknowledged and constrained by a sanity check that restricts the possible discrepancy in balance that Oracle can report.
+
+The current approach to sanity checking allows the Oracle committee to bring up to a 5% reduction of TVL in each report. Given that the governance reaction time has a lower bound of 72 hours, the malicious or compromised Oracles could reduce reported TVL by 15-20%, invoking mass liquidations on lending markets and dropping the price of stETH. However, a real negative rebase has very distinct features that we can take into account to reduce the impact and attack surface while allowing frictionless operation of the protocol even during a mass slashing event.
+
+While a high-precision check is possible using zero-knowledge technologies and EIP-4788, we decided to start with the simplier solution. It limits the impact in a simple but effective way. It privides an easy way to improve the robustness of the solution later.
+
+## Specification
+
+The proposed sanity check operates on the rebase value of the report
+$$
+clRebaseValue_i = 
+clBalance_i + withdrawalVaultBalance_i - clBalance_{i-1} - clDepositsAppeared_i
+$$
```

Comment:
This block has visual glitches in the GH preview window, while the formulas below look good

![image](https://github.com/lidofinance/lido-improvement-proposals/assets/961317/03b121ff-b5fa-4d21-b5e7-252482426bb6)